executers/populateandpublish.go

```diff
@@ -267,6 +267,9 @@ func (pwd *PackageWithDeps) getModPathInTemp(tempDir string) string {
 	moduleId := pwd.Dependency.GetId()
 	moduleInfo := strings.Split(moduleId, ":")
 	moduleInfo[0] = replaceExclamationMarkWithUpperCase(moduleInfo[0])
```

Comment:
I think that the name ```replaceExclamationMarkWithUpperCase``` is a bit too long...
I suggest we have two functions - one named ```goModEncode``` and one named ```goModDecode```.
Let's have the two functions next to each other in the same utility file.
We can also add a test that does encode and decode, and verifies that the outcome has not changed,

Reply:
Done I will add a test

Reply:
Done added a test